client/logical_port_client.go

```diff
@@ -0,0 +1,11 @@
+package client
+
+import (
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+type LogicalPortClient interface {
+	ListLogicalPorts(localVarOptionals map[string]interface{}) (manager.LogicalPortListResult, error)
```

Comment:
do we not require passing context here?

Reply:
I think passing context to use API client is an implementation detail which not necessary for the interface user